core/chains/evm/client/node.go

```diff
@@ -438,16 +488,18 @@ func (n *node) HeaderByNumber(ctx context.Context, number *big.Int) (header *typ
 	defer cancel()
 	lggr := n.newRqLggr(switching(n)).With("number", number)
 
-	lggr.Debug("RPC call: evmclient.Client#HeaderByNumber")
+	start := time.Now()
 	if n.http != nil {
 		header, err = n.http.geth.HeaderByNumber(ctx, number)
 		err = n.wrapHTTP(err)
 	} else {
 		header, err = n.ws.geth.HeaderByNumber(ctx, number)
 		err = n.wrapWS(err)
 	}
+	duration := time.Since(start)
 
 	n.logResult(lggr, err, "header", header)
+	n.logTiming(lggr, duration, err, n.getRpcDomain(), "HeaderByNumber")
```

Comment:
Would it not make sense to merge this into `logResult` to avoid double calls? You can add the method name in the logging too that way since you need to pass it anyway

Reply:
Ah I just noticed that the way it's done now the logging doesn't make much sense anymore, since the "RPC call: ..." log is going to be logged after the "RPC call success" log.

Maybe I should just put the duration/etc. fields on the "RPC call success/failure" logs instead, and leave the old "RPC call ..." log where it was. I'm just going to add the method to the "RPC call success" to provide more information in that one.